demos/ASP.Net Core/ITOps.ViewModelComposition/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace ITOps.ViewModelComposition
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void AddViewModelComposition(this IServiceCollection services, string assemblySearchPattern = "*ViewModelComposition*.dll")
+        {
+            var fileNames = Directory.GetFiles(AppContext.BaseDirectory, assemblySearchPattern);
+
+            var types = new List<Type>();
+            foreach (var fileName in fileNames)
+            {
+                var temp = AssemblyLoader.Load(fileName)
+                    .GetTypes()
+                    .Where(t =>
+                    {
+                        var typeInfo = t.GetTypeInfo();
+                        return !typeInfo.IsInterface 
```

Comment:
![image](https://user-images.githubusercontent.com/677704/28464944-0d0107e2-6e1f-11e7-83d1-d23e5ce1560d.png)

This should do it:

```c#
return !t.GetTypeInfo().IsAbstract && typeof(IRouteInterceptor).IsAssignableFrom(t);
```

If the type is an interface, then `IsAbstract` will return true, negating the first condition. Checking `!IsInterface` is redundant.